tests/cupy_tests/manipulation_tests/test_shape.py

```diff
@@ -176,6 +176,19 @@ def test_external_ravel(self, xp):
         a = a.transpose(2, 0, 1)
         return xp.ravel(a)
 
+    @testing.for_orders('CFA')
+    @testing.numpy_cupy_array_equal()
+    def test_ravel_non_contiguous(self, xp, order):
+        a = xp.array([1])
+        b = xp.broadcast_to(a, (10,))
```

Comment:
It would be more important to test `xp.arange(10)[::2].ravel(order)`.  The spec says `ravel` output is contiguous, which is stricter than the memory does not overlap.  I suggest:
- `test_ravel_non_contiguous` for `xp.arange(10)[::2]` and
- `test_ravel_broadcasted` for `xp.broadcast_to(xp.array([1]), 10)` (the current test)